slatedb/src/clock.rs

```diff
@@ -57,6 +97,82 @@ impl SystemClock for DefaultSystemClock {
         let elapsed = tokio::time::Instant::now().duration_since(self.initial_instant);
         system_time_from_millis(self.initial_ts + elapsed.as_millis() as i64)
     }
+
+    #[cfg(feature = "test-util")]
+    fn advance(self: Arc<Self>, duration: Duration) -> Pin<Box<dyn Future<Output = ()> + Send>> {
+        Box::pin(tokio::time::advance(duration))
+    }
+
+    fn sleep(self: Arc<Self>, duration: Duration) -> Pin<Box<dyn Future<Output = ()> + Send>> {
+        Box::pin(tokio::time::sleep(duration))
+    }
+
+    fn ticker(self: Arc<Self>, duration: Duration) -> SystemClockTicker {
+        SystemClockTicker::new(self, duration)
+    }
+}
+
+/// A mock system clock implementation that uses an atomic i64 to track time.
+/// The clock always starts at 0 (the Unix epoch). Time only advances when the
+/// `advance` method is called.
+#[derive(Debug)]
+#[cfg(feature = "test-util")]
+pub struct MockSystemClock {
+    /// The current timestamp in milliseconds since the Unix epoch.
+    /// Can be negative to represent a time before the epoch.
+    current_ts: AtomicI64,
+}
+
+#[cfg(feature = "test-util")]
+impl Default for MockSystemClock {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[cfg(feature = "test-util")]
+impl MockSystemClock {
+    pub fn new() -> Self {
+        Self {
+            current_ts: AtomicI64::new(0),
+        }
+    }
+
+    pub fn set(self: Arc<Self>, ts_millis: i64) {
+        self.current_ts.store(ts_millis, Ordering::SeqCst);
+    }
+}
+
+#[cfg(feature = "test-util")]
+impl SystemClock for MockSystemClock {
+    fn now(&self) -> SystemTime {
+        if self.current_ts.load(Ordering::SeqCst) < 0 {
```

Comment:
This isn't directly related to this PR, but my tests uncovered it.

Reply:
can you please add a comment with more details?

Reply:
Oops, I left this comment in the wrong spot. I meant to leave it here, which you also caught. :)

https://github.com/slatedb/slatedb/pull/683/files#r2213040943